lazy.go

```diff
@@ -51,7 +51,11 @@ type lazyConn struct {
 }
 
 func (l *lazyConn) Read(b []byte) (int, error) {
-	if !l.rhandshake {
+	l.rhlock.RLock()
+	rhandshake := l.rhandshake
+	l.rhlock.RUnlock()
```

Comment:
Why even do it in a goroutine? We need to wait for it to finish anyways. Can't we just have a `lazyInitialize` function and call `once.Do(msm.lazyInitialize)`?